lib/util.js

```diff
@@ -306,11 +306,23 @@ if (process.env.NODE_DEBUG) {
   debugEnvRegex = new RegExp(`^${debugEnv}$`, 'i');
 }
 
+// emits warning, when user sets
+// NODE_DEBUG=http or NODE_DEBUG=http2
+function emitWarningIfNeeded(set) {
+  if ('HTTP' === set || 'HTTP2' === set) {
+    process.emitWarning('Setting the NODE_DEBUG environment variable ' +
+      'to \'' + set.toLowerCase() + '\' can expose sensitive ' +
+      'data (including passwords, tokens, authentication headers) ' +
```

Comment:
Optional suggested text changes:

* s/including/such as/
*  s/tokens, authentication/tokens, and authentication/

Tests will need to be updated to reflect any changes here, of course.